floodlight/transforms/filter.py

```diff
@@ -0,0 +1,253 @@
+    This is a wrapper for the `scipy.filter.savgol
+    <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.butter.html>`__
+    function.
+
+    Parameters
+    ----------
+    xy: XY
+        Floodlight.XY-object.
+    window_length: int
+        The length of the filter window. This is the argument ``window_length`` from the
+        `scipy.filter.savgol <https://docs.scipy.org/doc/scipy/reference/generated/
+        scipy.signal.savgol_filter.html>`__ function. Default is 5.
+    poly_order: Numeric
+        The order of the polynomial used to fit the samples. ``poly_order`` must be less
+        than ``window_length``. Default is 3. This is the argument ``polyorder`` from
+        the `scipy.filter.savgol <https://docs.scipy.org/doc/scipy/reference/generated/
+        scipy.signal.savgol_filter.html>`__ function. Default is 5.
+    remove_short_seqs: bool
+        If True, sequences that are to short for the Filter with the specified settings
+        are removed from the data. If False, they are kept unfiltered. Default is True.
+
+    Returns
+    -------
+    xy_filtered: XY
+        XY object with position data filtered by designed Butterworth low pass filter.
+
+    Notes
+    -----
+    The values of the input data are assumed to be numerical. Missing data is assumed
+    to be either np.NaN or None. The Savitzky-Golay-filter requires a minimum signal
+    length depending on the settings. A signal is a sequence of data in the XY-object
+    that is not interrupted by missing values. The minimum signal length is defined as
+    the ``window_length``. The treatment of signals shorter than the minimum signal
+    length are specified with the ``remove_short_sequence``-argument, where True will
+    replace these sequences with np.NaNs ond False will keep the sequences in the data
+    unfiltered.
+
+    References
+    ----------
+        .. [2] `Savitzky, A.; Golay, M.J. (1964). Smoothing and differentiation of data
+            by simplified least squares procedures. Analytical Chemistry, 36(1), 1627-
+            1639. <https://pubs.acs.org/doi/abs/10.1021/ac60214a047>`_
+
+    """
+
+    min_signal_len = window_length
+
+    xy_filt = np.empty(xy.xy.shape)
+    for i, column in enumerate(xy.xy.T):
+        sequences = _get_sequences_without_nans(column)
+
+        col_filt = np.full(column.shape, np.nan)
+
+        for seq in sequences[sequences[:, 2] == 1]:
+            if np.diff(seq[0:2]) > min_signal_len:
+                col_filt[seq[0] : seq[1]] = scipy.signal.savgol_filter(
+                    column[seq[0] : seq[1]], window_length, poly_order
+                )
+            elif remove_short_seqs is False:
+                col_filt[seq[0] : seq[1]] = column[seq[0] : seq[1]]
+            else:
+                pass
+
+        xy_filt[:, i] = col_filt
+
+    xy_filtered = XY(xy=xy_filt, framerate=xy.framerate, direction=xy.direction)
+
+    return xy_filtered
```

Comment:
pretty much everything commented for the butterworth applies to the savgol as well